examples/nanocoap_server/coap_handler.c

```diff
@@ -182,6 +186,48 @@ NANOCOAP_RESOURCE(sha256) {
     .path = "/sha256", .methods = COAP_POST, .handler = _sha256_handler
 };
 
+/* separate response requires an event thread to execute it */
+#ifdef MODULE_EVENT_THREAD
+static nanocoap_sock_response_ctx_t _separate_ctx;
+static bool _separate_in_progress;
+
+static void _send_response(void *ctx)
+{
+    const char response[] = "This is a delayed response.";
+
+    puts("_separate_handler(): send delayed response");
+    nanocoap_sock_send_separate(ctx, COAP_CODE_CONTENT, COAP_TYPE_NON,
+                                response, sizeof(response));
+    _separate_in_progress = false;
+}
+
+static ssize_t _separate_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len, coap_request_ctx_t *context)
+{
+    static event_timeout_t event_timeout;
+    static event_callback_t event_timed = EVENT_CALLBACK_INIT(_send_response, &_separate_ctx);
+
+    if (_separate_in_progress) {
```

Comment:
Does this not imply a race condition between `event` and `coap` thread?

Reply:
I don't quite see it, the event thread can only set this to `false` when it's done.

Reply:
An unprotected static variable used in to threads always rings a bell in my head^^. But maybe nothing bad can happen if it is used like this. Anyway,  `event_timeout_is_pending()` might be an option?

Reply:
good idea!